arbiter/server.go

```diff
@@ -262,16 +279,27 @@ func (s *Server) loadStatus() (int, error) {
 	return status, errors.Trace(err)
 }
 
-func syncBinlogs(source <-chan *reader.Message, ld loader.Loader) {
+func syncBinlogs(source <-chan *reader.Message, ld loader.Loader, res chan error) {
 	dest := ld.Input()
+	var txn *loader.Txn
+	var err error = nil
 	for msg := range source {
 		log.Debug("recv msg from kafka reader", zap.Int64("ts", msg.Binlog.CommitTs), zap.Int64("offset", msg.Offset))
-		txn := loader.SlaveBinlogToTxn(msg.Binlog)
+		if err != nil {
+			continue
+		}
+		txn, err = loader.SlaveBinlogToTxn(msg.Binlog)
+		if err != nil {
+			log.Error("transfer binlog failed, program will stop handling data from loader", zap.Error(err))
+			res <- err
+			continue
+		}
```

Comment:
It seems like we could just return the error here. If I didn't misunderstand the code above, after the first error, all messages would simply be skipped, so unless we had to read all the messages from the channel, we can just return.

Reply:
According to @lichunzhu , the goroutine running `kafkaReader` may block waiting to send message to a full channel if we just return here.
If `kafkaReader` is closed, the the goroutine would panic because it attempts to send to a closed channel.
Anyhow, there's space of improvement in the implementation of the `reader`, so @lichunzhu  will take care to fix `reader` first.

Reply:
Now this function will return immediately